scheduler/conflicts.py

```diff
@@ -0,0 +1,64 @@
+"""module responsible fro finding conflicts in database"""
+from typing import List, Tuple
+from enum import Enum
+from scheduler.models import Lesson, Professor, Auditorium, Group
+
+class ConflictType(Enum):
+    PROFESSOR = 1
+    AUDITORIUM = 2
+    GROUP = 3
+
+def check_time_conflict(lesson1: Lesson, lesson2: Lesson) -> bool:
```

Comment:
Proponowałbym nazwę typu `are_overlapping` - wyraźnie sygnalizuje zwracanie boolea i że nie ma tu żadnej głębszej magii. Zastanawiam się też, czy warto zamieszczać fragment `Lessons should have common factor like same professor, auditorium, group` w docu - to nie jest konieczne żeby ta funckja poprawnie zadziałała.